### PR TITLE
[discovery][dns] move dns to own package

### PR DESCRIPTION
**Pull request by stuartnelson3**
@fabxc 

- move dns to own package
- add additional select{} wrapping channel send retrieval/discovery/dns/dns.go:161
- remove unused done struct chan from dns discovery struct

### CODE REVIEW

File: retrieval/discovery/dns/dns.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+				log.Errorf("Error refreshing DNS targets: %s", err)
+			}
+			wg.Done()
+		}(name)
+	}
+
+	wg.Wait()
+}
+
+func (dd *Discovery) refresh(name string, ctx context.Context, ch chan<- []*config.TargetGroup) error {
```
Comment (fabxc) [inline retrieval/discovery/dns/dns.go]: So far we are passing `ctx` as the first argument consistently.
Let's do this here as well.

**Comment by fabxc** [comment]
Looks good aside from the argument order.
Please split pure moving of code into separate commits next time :) easier to review the actual change.

File: retrieval/discovery/dns/dns.go (hunk 1 above)
Comment (stuartnelson3) [inline retrieval/discovery/dns/dns.go]: cool, pushed

**Comment by fabxc** [comment]
Can you prefix these commits with `discovery/dns: ...`.
If we are consistent with that we can generate changelogs based on commits for future releases.

**Comment by fabxc** [comment]
Thanks

**Pull request merged**